iroh-sync/src/sync.rs

```diff
@@ -1297,6 +1395,64 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_prefix_delete_memory() -> Result<()> {
+        let store = store::memory::Store::default();
+        test_prefix_delete(store)?;
+
+        Ok(())
+    }
+
+    #[cfg(feature = "fs-store")]
+    #[test]
+    fn test_prefix_delete_fs() -> Result<()> {
+        let dbfile = tempfile::NamedTempFile::new()?;
+        let store = store::fs::Store::new(dbfile.path())?;
+        test_prefix_delete(store)?;
+        Ok(())
+    }
+
+    #[test]
+    fn test_insert_empty() -> Result<()> {
+        let store = store::memory::Store::default();
+        let mut rng = rand::thread_rng();
+        let alice = Author::new(&mut rng);
+        let myspace = Namespace::new(&mut rng);
+        let replica = store.new_replica(myspace.clone())?;
+        let hash = Hash::new(b"");
+        let res = replica.insert(b"foo", &alice, hash, 0);
+        assert!(matches!(res, Err(InsertError::EntryIsEmpty)));
+        Ok(())
+    }
+
+    fn test_prefix_delete<S: store::Store>(store: S) -> Result<()> {
+        let mut rng = rand::thread_rng();
+        let alice = Author::new(&mut rng);
+        let myspace = Namespace::new(&mut rng);
+        let replica = store.new_replica(myspace.clone())?;
+        let hash1 = replica.hash_and_insert(b"foobar", &alice, b"hello")?;
+        let hash2 = replica.hash_and_insert(b"fooboo", &alice, b"world")?;
+
+        // sanity checks
+        assert_eq!(
+            get_content_hash(&store, myspace.id(), alice.id(), b"foobar")?,
+            Some(hash1)
+        );
+        assert_eq!(
+            get_content_hash(&store, myspace.id(), alice.id(), b"fooboo")?,
+            Some(hash2)
+        );
+
+        // delete
+        let deleted = replica.wipe_at_prefix(b"foo", &alice)?;
+        assert_eq!(deleted, 2);
+        assert_eq!(store.get_one(myspace.id(), alice.id(), b"foobar")?, None);
+        assert_eq!(store.get_one(myspace.id(), alice.id(), b"fooboo")?, None);
+        assert_eq!(store.get_one(myspace.id(), alice.id(), b"foo")?, None);
+
+        Ok(())
+    }
```

Comment:
we will need some more extenstive tests here, especially for the db part, making sure we don’t delete the wrong things

Reply:
Do you have some specific test cases in mind here?

Reply:
mostly around the prefix logic, edge cases etc adding and removing things a lot